app/topics/topic-management/add-existing-subtopic/add-existing-subtopic.component.ts

```diff
@@ -36,19 +38,31 @@ export class AddExistingSubtopicComponent {
     this.showChange.emit(this.show);
   }
 
-  public searchTopics() {
+  private filterTopics(topics:any) {
+    let i =0;
+    let j=0;
```

Comment:
you don't need to initialize a variable before a loop, you can do it in the loop directly